benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,319 @@
-    
-
-    def _cross_entropy_tensor():
-        torch.nn.functional.cross_entropy(a, b)
-    return _cross_entropy_tensor
-
-def cross_entropy_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    targets = [torch.randint(1, (H, W), dtype=torch.int64) for _ in range(N)]
-
-    a = nestedtensor.nested_tensor(inputs)
-    b = nestedtensor.nested_tensor(targets)
-
-    def _cross_entropy_nt():
-        torch.nn.functional.cross_entropy(a, b)
-    return _cross_entropy_nt
-
-#
-# dropout
-#
-def dropout_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _dropout_tensor():
-        torch.nn.functional.dropout(a)
-    return _dropout_tensor
-
-def dropout_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _dropout_nt():
-        torch.nn.functional.dropout(nt)
-    
-    return _dropout_nt
-
-#
-# interpolate
-#
-def interpolate_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _interpolate_tensor():
-        torch.nn.functional.interpolate(a, inputs[0].unsqueeze(0).shape[-2])
-    return _interpolate_tensor
-
-def interpolate_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _interpolate_nt():
-        torch.nn.functional.interpolate(nt)
-    
-    return _interpolate_nt
+class SegLayersBenchMark(object):
+    def __init__(self, args):
+        assert len(args) == 8
+        print("Called with args: ")
+        
+        self.N = int(args[0])
+        print("N = ", self.N)
 
+        self.C = int(args[1])
+        print("C = ", self.C)
 
-if __name__ == "__main__":
-    print(utils.benchmark_fn(relu_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(relu_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(conv2d_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(conv2d_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(batch_norm_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(batch_norm_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_nt_tensors_same_size(), warmup=2.0))
+        self.H = int(args[2])
+        print("H = ", self.H)
```

Comment:
Instead of printing it, it'd probably be better to repeat per experiment.

Reply:
what do you mean repeat?